File: CS/CustomDescendants/MyGridControl.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Registrator;
using DevExpress.XtraGrid.Dragging;

namespace TreeListCustomization
{
    class MyGridControl : GridControl
    {
        protected override BaseView CreateDefaultView() { return CreateView("MyBandedGridView"); }
        protected override void RegisterAvailableViewsCore(InfoCollection collection)
        {
            base.RegisterAvailableViewsCore(collection);
            collection.Add(new MyBandedGridViewInfoRegistrator());
            collection.Add(new MyAdvBandedGridViewInfoRegistrator());
        }

        internal new DragController DragController { get { return base.DragController; } }
    }

}

File: VB/CustomDescendants/MyGridControl.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Registrator
Imports DevExpress.XtraGrid.Dragging

Namespace TreeListCustomization
	Friend Class MyGridControl
		Inherits GridControl
		Protected Overrides Function CreateDefaultView() As BaseView
			Return CreateView("MyBandedGridView")
		End Function
		Protected Overrides Sub RegisterAvailableViewsCore(ByVal collection As InfoCollection)
			MyBase.RegisterAvailableViewsCore(collection)
			collection.Add(New MyBandedGridViewInfoRegistrator())
			collection.Add(New MyAdvBandedGridViewInfoRegistrator())
		End Sub

		Friend Shadows ReadOnly Property DragController() As DragController
			Get
				Return MyBase.DragController
			End Get
		End Property
	End Class

End Namespace
